File: ejemplos/avanzado/benchmark_eratosthenes/eratosthenes.nim
import math

proc eratosthenes(n: int): auto =
  result = newSeq[int8](n + 1)
  result[0] = 1
  result[1] = 1

  for i in 0 .. sqrt(float(n)).int:
    if result[i] == 0:
      for j in countup(i * i, n, i):
        result[j] = 1


discard eratosthenes(100_000_000)

File: ejemplos/avanzado/benchmark_eratosthenes/eratosthenes.py

def eratosthenes(n: int) -> list:
  sieve = [1] * 2 + [0] * (n - 1)

  for i in range(int(n ** 0.5)):
    if not sieve[i]:
      for j in range(i * i, n + 1, i):
        sieve[j] = 1

  return sieve


eratosthenes(100_000_000)
